internal/services/migrate_rewards_test.go

```diff
@@ -0,0 +1,97 @@
+package services
+
+import (
+	"context"
+	"testing"
+
+	pb_defs "github.com/DIMO-Network/device-definitions-api/pkg/grpc"
+	"github.com/DIMO-Network/rewards-api/internal/utils"
+	"github.com/DIMO-Network/rewards-api/models"
+	"github.com/ericlagergren/decimal"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/types"
+)
+
+func TestMigrateOldRewards(t *testing.T) {
+	ctx := context.Background()
+
+	logger := zerolog.Nop()
+
+	cont, conn := utils.GetDbConnection(ctx, t, logger)
+	defer func() {
+		err := cont.Terminate(ctx)
+		assert.NoError(t, err)
+	}()
+
+	fkc := &FakeDefClient{}
+	intgs, err := fkc.GetIntegrations(ctx, nil, nil)
+	assert.NoError(t, err)
+
+	integrsByID := make(map[string]*pb_defs.Integration)
+	for _, integr := range intgs.Integrations {
+		integrsByID[integr.Id] = integr
+	}
+
+	issuanceWeeks := []models.IssuanceWeek{
+		{ID: 1, JobStatus: models.IssuanceWeeksJobStatusFinished},
+	}
+
+	for _, isw := range issuanceWeeks {
+		err := isw.Insert(ctx, conn.DBS().Writer, boil.Infer())
+		assert.NoError(t, err)
+	}
+
+	existingRwrds := []models.Reward{
+		{
+			IssuanceWeekID:      1,
+			UserDeviceID:        "MockDeviceID1",
+			UserID:              "User1",
+			ConnectionStreak:    13,
+			DisconnectionStreak: 0,
+			StreakPoints:        2000,
+			IntegrationPoints:   int(integrsByID[autoPiIntegration].Points) + int(integrsByID[teslaIntegration].Points),
+			Tokens:              types.NewNullDecimal(decimal.New(int64(7000), 0)),
+			UserEthereumAddress: null.StringFrom(mkAddr(1).Hex()),
+			TransferSuccessful:  null.BoolFrom(true),
+			UserDeviceTokenID:   types.NewNullDecimal(decimal.New(int64(1), 0)),
+			IntegrationIds: types.StringArray{
+				autoPiIntegration,
+				teslaIntegration,
+			},
+		},
+	}
+
+	for _, rwrd := range existingRwrds {
+		err := rwrd.Insert(ctx, conn.DBS().Writer, boil.Infer())
+		assert.NoError(t, err)
+	}
+
+	err = MigrateRewardsController(ctx, &logger, conn, intgs, 2)
+	assert.NoError(t, err)
+
+	rewards, err := models.Rewards().All(ctx, conn.DBS().Reader)
+	assert.NoError(t, err)
+
+	expected := models.Reward{
```

Comment:
I like the test. I'd be interested to see

* Multiple weeks
* Only one method
* Streak points 0

Reply:
Let's at least modify _this_ test to have

* More than one row
* A `tokens` amount that doesn't fit in `integer`

Since these were the two big bugs we squashed.